### PR TITLE
Remove the PVs for maven caching

### PR DESCRIPTION
**Pull request by jaseemabid**
Fixes https://github.com/openshiftio/openshift.io/issues/2368

### CODE REVIEW

File: vars/mavenTemplate.groovy [hunk 1]
```diff
@@ -9,7 +9,7 @@ def call(Map parameters = [:], body) {
     def defaultLabel = buildId('maven')
     def label = parameters.get('label', defaultLabel)
 
-    def mavenImage = parameters.get('mavenImage', 'fabric8/maven-builder:vd81dedb')
+    def mavenImage = parameters.get('mavenImage', 'fabric8/maven-builder:SNAPSHOT-PR-33-1')
```
Comment (chmouel) [inline vars/mavenTemplate.groovy]: why do you use the tagged snapshot here instead of revision?

**Comment by chmouel** [comment]
Have you been  able to test this change properly, can you send us a trail log with numbers before and after please? (and due show us that works properly)

File: vars/mavenTemplate.groovy (hunk 1 above)
Comment (jaseemabid) [reply vars/mavenTemplate.groovy]: @chmouel I'm using this PR to get a build going. I'll use revisions before the final revision of the PR.

File: vars/mavenBuildSnapshot.groovy [hunk 2]
```diff
@@ -15,7 +15,7 @@ def call(body) {
         // update any versions that we want to override
         overwriteDeps(config.pomVersionToUpdate)
 
-        sh "mvn clean -e -U install -Dm2.repo=.mvnrepository -Duser.home=/root/ -Ddocker.push.registry=${env.FABRIC8_DOCKER_REGISTRY_SERVICE_HOST}:${env.FABRIC8_DOCKER_REGISTRY_SERVICE_PORT}"
+        sh "mvn clean -e -U install -Ddocker.push.registry=${env.FABRIC8_DOCKER_REGISTRY_SERVICE_HOST}:${env.FABRIC8_DOCKER_REGISTRY_SERVICE_PORT}"
```
Comment (hrishin) [inline vars/mavenBuildSnapshot.groovy]: @jaseemabid I think we need to add `Dm2.repo`, `Duser.home` switches back for respective location

**Comment by chmouel** [comment]
@jaseemabid I have reverted some part into 7fdbffe  to address @hrishin comments, so we can merge this quickly, if you feel it's still needed feel free to send a new pr,

**Comment by jfchevrette** [comment]
Tested on starter-us-east-2a and build slaves are not using a PV anymore.

**Comment by jfchevrette** [comment]
Confirmed working on starter-us-east-2 as well

**Comment by jaseemabid** [comment]
Thanks for merging @chmouel.

**Pull request merged**